okhttp/src/main/java/okhttp3/RealCall.java

```diff
@@ -33,7 +33,10 @@
 final class RealCall implements Call {
   final OkHttpClient client;
   final RetryAndFollowUpInterceptor retryAndFollowUpInterceptor;
-  final EventListener eventListener;
+
+  // There is a cycle between the Call and EventListener that makes this awkward. This will be set
+  // after we create the call instance then create the event listener instance.
+  volatile EventListener eventListener;
```

Comment:
(the RealCall itself is safely published later when we share it with more threads)